src/test/librados/service.cc

```diff
@@ -114,3 +114,45 @@ TEST(LibRadosServicePP, Status) {
   }
   cluster.shutdown();
 }
+
+TEST(LibRadosServicePP, Close) {
+  int tries = 20;
+  string name = string("close-test-pid") + stringify(getpid());
+  int i;
+  for (i = 0; i < tries; ++i) {
+    cout << "attempt " << i << " of " << tries << std::endl;
+    {
+      Rados cluster;
+      cluster.init("admin");
+      ASSERT_EQ(0, cluster.conf_read_file(NULL));
+      cluster.conf_parse_env(NULL);
+      ASSERT_EQ(0, cluster.connect());
+      ASSERT_EQ(0, cluster.service_daemon_register(
+		  "laundry", name, {{"foo", "bar"}, {"this", "that"}}));
+      sleep(3); // let it register
+      cluster.shutdown();
+    }
+    // mgr updates servicemap every tick
+    sleep((int)g_conf->mgr_tick_period);
```

Comment:
mgr_tick_period is a new-style setting, and can only retrieved `g_conf->get_val<std::chrono::seconds>("mgr_tick_period")`. i'd suggest change this part of something like
```c++
std::this_thread::sleep_for(g_conf->get_val<std::chrono::seconds>("mgr_tick_period"));
```

```
/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos7/DIST/centos7/MACHINE_SIZE/huge/release/13.0.2-1086-gd04816a/rpm/el7/BUILD/ceph-13.0.2-1086-gd04816a/src/test/librados/service.cc: In member function 'virtual void LibRadosServicePP_Close_Test::TestBody()':
/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos7/DIST/centos7/MACHINE_SIZE/huge/release/13.0.2-1086-gd04816a/rpm/el7/BUILD/ceph-13.0.2-1086-gd04816a/src/test/librados/service.cc:136:24: error: 'struct md_config_t' has no member named 'mgr_tick_period'; did you mean 'mon_tick_interval'?
     sleep((int)g_conf->mgr_tick_period);
                        ^~~~~~~~~~~~~~~
                        mon_tick_interval
```

see https://jenkins.ceph.com/job/ceph-dev-new-build/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=centos7,DIST=centos7,MACHINE_SIZE=huge/10875//consoleFull

Reply:
oops, fixed!  teh std::this_thread thing didn't compile so i just did normal sleep().

Reply:
```diff
diff --git a/src/test/librados/service.cc b/src/test/librados/service.cc
index 2720bbd..c76bd93 100644
--- a/src/test/librados/service.cc
+++ b/src/test/librados/service.cc
@@ -5,6 +5,7 @@
 #include "include/stringify.h"

 #include <algorithm>
+#include <thread>
 #include <errno.h>
 #include "gtest/gtest.h"
 #include "test/unit.cc"
@@ -133,7 +134,7 @@ TEST(LibRadosServicePP, Close) {
       cluster.shutdown();
     }
     // mgr updates servicemap every tick
-    sleep(g_conf->get_val<int64_t>("mgr_tick_period"));
+    std::this_thread::sleep_for(g_conf->get_val<std::chrono::seconds>("mgr_tick_period"));
     // make sure we are deregistered
     {
       Rados cluster;
```

@liewegas might want to apply the patch above. otherwise we will have http://pulpito.ceph.com/kchai-2018-04-21_23:40:39-rados-wip-kefu-testing-2018-04-22-0053-distro-basic-mira/2425416/

```
2018-04-21T23:57:13.852 INFO:tasks.workunit.client.0.mira012.stdout:              api_service: [ RUN      ] LibRadosServicePP.Close
2018-04-21T23:57:13.852 INFO:tasks.workunit.client.0.mira012.stdout:              api_service: attempt 0 of 20
2018-04-21T23:57:13.852 INFO:tasks.workunit.client.0.mira012.stdout:              api_service: unknown file: Failure
2018-04-21T23:57:13.852 INFO:tasks.workunit.client.0.mira012.stdout:              api_service: C++ exception with description "boost::bad_get: failed value ge
t using boost::get" thrown in the test body.
2018-04-21T23:57:13.852 INFO:tasks.workunit.client.0.mira012.stdout:              api_service: [  FAILED  ] LibRadosServicePP.Close (3027 ms)
```